src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const item of items) {
+    const value: string | number = item[key];
+
+    if (result[value] !== undefined) {
+      result[value].push(item);
+    }
+
+    if (result[value] === undefined) {
+      result[value] = [];
+      result[value].push(item);
+    }
```

Comment:
```suggestion
    if (!result[value]) {
      result[value] = [];
    }
    result[value].push(item);
```

we can simplify it a bit